tests/scripts/runincontext.cmd

```diff
@@ -0,0 +1,3 @@
+set CORE_LIBRARIES=%~dp1
```

Comment:
These kinds of scrips are difficult to understand and people spent more time trying to understand all the details and the intent rather than making progress. Can you comment usage and what this script expects?

For example, why `%~dp1`? This implies the first argument is a path to where `CORE_LIBRARIES` is? What about the remaining args? Is `%1%2` right? Should it have a space?

Some doc at the top of this script should be added for context and intent.

Reply:
I will add comments. This is based on how the script is called by the .cmd of the individual tests if they find the `CLRCustomTestLauncher` is defined. Btw, I have not added this feature, we have had it before. 
This is what we have in the test `.cmd` files:
```bat
IF NOT "%CLRCustomTestLauncher%"=="" (
  set LAUNCHER=call %CLRCustomTestLauncher% %~dp0
) ELSE (
  set LAUNCHER=%_DebuggerFullPath% "%CORE_ROOT%\corerun.exe"
)
```
and then
```bat
%LAUNCHER% %ExePath% %CLRTestExecutionArguments%
```

Here is an example how a test calls that script
```
call D:\git\coreclr2\tests\scripts\runincontext.cmd D:\git\coreclr\bin\tests\Windows_NT.x64.Debug\GC\Features\LOHCompaction\lohpin\ lohpin.exe
```

So the `%1%2` forms the absolute path to the test executable. The `%~dp1` is meant to be the path to the test directory. It could actually be just %1, as it is equivalent in this case.